lib/OpenQA/WebSockets/Client.pm

```diff
@@ -29,29 +29,25 @@ my $IS_WS_SERVER_ITSELF;
 sub mark_current_process_as_websocket_server { $IS_WS_SERVER_ITSELF = 1; }
 sub is_current_process_the_websocket_server  { return $IS_WS_SERVER_ITSELF; }
 
-sub new {
-    my $class = shift;
+sub new ($class) {
```

Comment:
```suggestion
sub new ($class, @args) {
```
and then later
```
$class->SUPER::new(@args);
```

Reply:
ok, cool. Could you explain the error message to me? Why does it say "Can't use string ("OpenQA::WebSockets::Client") as a HASH ref"?

Reply:
Before it was doing `my $class = shift`, so `"OpenQA::WebSockets::Client"` wasn't in `@_` anymore.
Now with the missing shift, it was passed as the second arg to `$class->SUPER::new(@_);`

Reply:
Ah, I see. Strange that even with `Carp::Always` I could not see any trace mention that line to be problematic

Reply:
anyway, fixed now thanks to your suggestion.